opencti-platform/opencti-front/src/private/components/settings/sub_types/EntitySettingScale.tsx

```diff
@@ -0,0 +1,221 @@
+import makeStyles from '@mui/styles/makeStyles';
+import React, { useState, FunctionComponent } from 'react';
+import * as Yup from 'yup';
+import { Add } from '@mui/icons-material';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import { Button } from '@mui/material';
+import { useFormatter } from '../../../../components/i18n';
+import { MESSAGING$ } from '../../../../relay/environment';
+import type { ScaleConfig, Tick, UndefinedTick } from './EntitySettingAttributesConfigurationScale';
+import EntitySettingScaleTickLine from './EntitySettingScaleTickLine';
+
+const useStyles = makeStyles(() => ({
+  container: {
+    padding: '10px 20px 20px 20px',
+  },
+}));
+
+interface EntitySettingScaleProps {
+  scaleConfig: ScaleConfig,
+  submitting: boolean,
+  handleSubmit: (scale: ScaleConfig) => void
+}
+
+const isTickDefinitionValid = (tickDefinition: ScaleConfig) => {
+  let isValid = true;
+  if (tickDefinition.min.value > tickDefinition.max.value) {
+    MESSAGING$.notifyError('The min value cannot be greater than max value');
+    return false;
+  }
+  tickDefinition.ticks.forEach((tick) => {
+    if (!!tick.value && (tick.value < tickDefinition.min.value || tick.value > tickDefinition.max.value)) {
```

Comment:
If you add a tick, it doesn't have a value at first. 
If we remove the check, we get an error ```'tick.value' is possibly 'undefined'```

Reply:
I hear you, yet I don't think we should consider an undefined tick value as a "valid" possibility. You may want to add another check before this condition

Reply:
I don't see how I could add another check before the condition, if I remove `!!tick.value`, the rest of my condition doesn't work. Do you have a suggestion that works ?

Reply:
> Tests API :
> 
> * The available options are not manage when I get the attributesDefinitions property. I should not have the scale of the confidence property.
>   ![image](https://user-images.githubusercontent.com/53513584/224651388-3c367b35-ebeb-4465-88f9-3032e89c1a42.png)

---> You ask for the scale in your query. If you don't ask for it, you won't have the scale

> * What's the effects of changing better_side property ?
---> It's not used for now but was requested for an eventual V2

> * I can have min equals to max
---> The min cannot be greater than max but that's the only condition

> 
> Tests front :
> 
> * I have a missing translation on confidence scale. And the Low/faible it's translated but it's a user configuration so.. I am not sure it's a good idea to translate it
>   ![image](https://user-images.githubusercontent.com/53513584/224659309-beb05d1c-ef68-495e-a730-8a5e1d9c6f15.png)

--> same answer as for Yassine. We need the default label to be translated, but we can't translate a custom one because we cannot know in advance what it will be. The probability to be the same as words translated in the platform are very very low.

> * I can't fill the confidence field to null

--> Yes it's not possible for now, it will be in a V2 (we plan to open issues after the merge)